GoogleForCreators/web-stories-wp
# UI Simplification: Consider aspect ratio lock when resizing

## Summary
Previously, the aspect ratio lock of the element (configured in the design panel) and resizing the element on canvas were not related.

With this PR, the configured lock will also influence resizing the element. If the ratio is locked, it's also locked when resizing, when unlocked, it's unlocked (even from the corners).

New images and rectangles will now have unlocked aspect ratio by default.
Holding down `Shift` key will force lock, no matter the design panel setting.


## Relevant Technical Choices



## To-do




## User-facing changes
Text resize: unlocked and then locked (with `Shift`)

![text-resize](https://user-images.githubusercontent.com/3294597/151156028-9d603df4-364a-4500-8232-0cbf89f2ceaa.gif)

Image resize: unlocked and then locked (with `Shift`)
![image-resize](https://user-images.githubusercontent.com/3294597/151156065-29c20ee0-ce46-48b2-ba64-8c1955279564.gif)



## Testing Instructions






This PR can be tested by following these steps:

1. Verify that a newly added rectangle or image has unlocked aspect ratio
2. Verify that resizing any element with unlocked aspect ratio (from corners as well) indeed resizes it without keeping ratio (note: stickers can't have unlocked ratio).
3. Verify that resizing from any side with locked ratio keeps the ratio locked.
4. Verify that when text ratio is unlocked, it can only be resized horizontally.
5. Verify that holding down "Shift" forces resizing with keeping ratio, independent of the design panel setting.


## Reviews

### Does this PR have a security-related impact?
No


### Does this PR change what data or activity we track or use?
No


### Does this PR have a legal-related impact?
No


## Checklist




---



Fixes #10106
